### PR TITLE
Update RSPEC

### PR DESCRIPTION
**Pull request by pavel-mikula-sonarsource**
Update RSPEC before release

This is first RSPEC update after rule-api v2.0 release

I did some manual changes, please re-run the rule-api to see and validate them.

### CODE REVIEW

File: analyzers/rspec/cs/S2326_c#.json [hunk 1]
```diff
@@ -12,5 +12,10 @@
   "defaultSeverity": "Major",
   "ruleSpecification": "RSPEC-2326",
   "sqKey": "S2326",
-  "scope": "All"
+  "scope": "All",
+  "securityStandards": {
+    "CERT": [
+      "MSC12-CPP."
+    ]
```
Comment (costin-zaharia-sonarsource) [inline analyzers/rspec/cs/S2326_c#.json:19]: Isn't this wrong? To have a cpp standard on a C# project?

Comment (pavel-mikula-sonarsource) [reply analyzers/rspec/cs/S2326_c#.json]: CERT removed manually.
There might be some more of them going away after RULEAPI-665 is fixed in further releases. There should be no harm, as they aren't user facing according to slack message.

File: analyzers/rspec/cs/S1117_c#.json [hunk 2]
```diff
@@ -13,5 +13,7 @@
   "defaultSeverity": "Major",
   "ruleSpecification": "RSPEC-1117",
   "sqKey": "S1117",
-  "scope": "All"
+  "scope": "All",
+  "securityStandards": {
+  }
```
Comment (costin-zaharia-sonarsource) [inline analyzers/rspec/cs/S1117_c#.json:18]: Would be great to have the tool updated to skip the generation of empty sections.

File: analyzers/rspec/cs/S2386_c#.html [hunk 3]
```diff
@@ -33,7 +33,7 @@ <h2>Exceptions</h2>
       <li> <code>System.Collections.Immutable.IImmutableSet&lt;T&gt;</code> </li>
       <li> <code>System.Collections.Immutable.IImmutableStack&lt;T&gt;</code> </li>
       <li> <code>System.Collections.Immutable.IImmutableQueue&lt;T&gt;</code> </li>
-    </ul> </li>
+    </ul>  </li>
```
Comment (costin-zaharia-sonarsource) [inline analyzers/rspec/cs/S2386_c#.html:36]: Strange that we get an extra space here.

File: analyzers/rspec/cs/S4581_c#.html [hunk 4]
```diff
@@ -10,6 +10,8 @@ <h2>Noncompliant Code Example</h2>
 public void Foo()
 {
     var g = new Guid(); // Noncompliant - what's the intent?
+    var guid = default(Guid);
+    Guid other = default;
```
Comment (costin-zaharia-sonarsource) [inline analyzers/rspec/cs/S4581_c#.html:14]: We don't support this yet. Needs to be reverted.

File: analyzers/rspec/cs/S1117_c#.json (hunk 2 above)
Comment (pavel-mikula-sonarsource) [reply analyzers/rspec/cs/S1117_c#.json:18]: That was probably just me, with manual edits.

File: analyzers/src/SonarAnalyzer.Utilities/Rules.Description/S4581.html [hunk 5]
```diff
@@ -10,6 +10,8 @@ <h2>Noncompliant Code Example</h2>
 public void Foo()
 {
     var g = new Guid(); // Noncompliant - what's the intent?
+    var guid = default(Guid);
+    Guid other = default;
```
Comment (costin-zaharia-sonarsource) [inline analyzers/src/SonarAnalyzer.Utilities/Rules.Description/S4581.html:14]: Here also. Probably this one will get fixed automatically when the other one is fixed.

File: analyzers/src/SonarAnalyzer.Utilities/Rules.Description/S4784_cs.html [hunk 6]
```diff
@@ -85,6 +85,4 @@ <h2>See</h2>
   </li>
   <li> OWASP Regular expression Denial of Service - ReDoS </li>
 </ul>
-<h2>Deprecated</h2>
-<p>This rule is deprecated; use {rule:csharpsquid:S2631} instead.</p>
```
Comment (costin-zaharia-sonarsource) [inline analyzers/src/SonarAnalyzer.Utilities/Rules.Description/S4784_cs.html:89]: I think we should keep this message. It's relevant for the user. Another tool issue?

**Review by costin-zaharia-sonarsource — CHANGES_REQUESTED** [review]
Please see my comments

File: analyzers/rspec/cs/S1117_c#.json (hunk 2 above)
Comment (costin-zaharia-sonarsource) [reply analyzers/rspec/cs/S1117_c#.json:18]: Ah, this appears in a lot of places.

File: analyzers/rspec/cs/S2386_c#.html (hunk 3 above)
Comment (pavel-mikula-sonarsource) [reply analyzers/rspec/cs/S2386_c#.html:36]: Source: https://raw.githubusercontent.com/SonarSource/rspec/master/rules/S2386/csharp/rule.adoc

Seems just like the way adoc serializes these nested bulletpoints.

File: analyzers/rspec/cs/S4581_c#.html (hunk 4 above)
Comment (pavel-mikula-sonarsource) [reply analyzers/rspec/cs/S4581_c#.html]: Nice catch!

File: analyzers/src/SonarAnalyzer.Utilities/Rules.Description/S4784_cs.html (hunk 6 above)
Comment (pavel-mikula-sonarsource) [reply analyzers/src/SonarAnalyzer.Utilities/Rules.Description/S4784_cs.html:89]: This rule was marked as `ready` during migration from Jira (see this [thread](https://sonarsource.slack.com/archives/CFSLVKNBB/p1624024528158900) ) while it should remain deprecated. So the tool itself is probably fine.

I just missed the fact that this should be undone.

File: analyzers/rspec/cs/S1117_c#.json (hunk 2 above)
Comment (pavel-mikula-sonarsource) [reply analyzers/rspec/cs/S1117_c#.json]: I just rerun that 6 LOC script I used to remove the CERT :)

**Review by costin-zaharia-sonarsource — APPROVED** [review]
LGTM!

**Pull request merged**